service/controller/v17/encrypter/kms/kms.go

```diff
@@ -44,72 +45,146 @@ func NewEncrypter(c *EncrypterConfig) (*Encrypter, error) {
 	return kms, nil
 }
 
-func (k *Encrypter) CreateKey(ctx context.Context, customObject v1alpha1.AWSConfig, keyAlias string) error {
-	sc, err := controllercontext.FromContext(ctx)
+func (e *Encrypter) EnsureCreatedEncryptionKey(ctx context.Context, customObject v1alpha1.AWSConfig) error {
+	ctlCtx, err := controllercontext.FromContext(ctx)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	key, err := sc.AWSClient.KMS.CreateKey(&kms.CreateKeyInput{})
-	if err != nil {
-		return microerror.Mask(err)
+	{
+		e.logger.LogCtx(ctx, "level", "debug", "message", "finding out encryption key")
+
+		_, err := e.describeKey(ctx, customObject)
+		if IsKeyNotFound(err) {
+			e.logger.LogCtx(ctx, "level", "debug", "message", "did not find encryption key")
+
+		} else if err != nil {
+			return microerror.Mask(err)
+
+		} else {
+
+			e.logger.LogCtx(ctx, "level", "debug", "message", "found encryption key")
+			e.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			return nil
+		}
 	}
 
-	_, err = sc.AWSClient.KMS.CreateAlias(&kms.CreateAliasInput{
-		AliasName:   aws.String(keyAlias),
-		TargetKeyId: key.KeyMetadata.Arn,
-	})
-	if err != nil {
-		return microerror.Mask(err)
+	var keyID *string
+	{
+		e.logger.LogCtx(ctx, "level", "debug", "message", "creating encryption key")
+
+		tags := key.ClusterTags(customObject, e.installationName)
+
+		// TODO already created case should be handled here. Otherwise there is a chance alias wasn't created yet and EncryptionKey will tell there is no key. We can check tags to see if the key was created. Issue: https://github.com/giantswarm/giantswarm/issues/4262.
+		in := &kms.CreateKeyInput{
+			Tags: awstags.NewKMS(tags),
+		}
+
+		out, err := ctlCtx.AWSClient.KMS.CreateKey(in)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		keyID = out.KeyMetadata.KeyId
+
+		e.logger.LogCtx(ctx, "level", "debug", "message", "created encryption key")
 	}
 
-	_, err = sc.AWSClient.KMS.EnableKeyRotation(&kms.EnableKeyRotationInput{
-		KeyId: key.KeyMetadata.KeyId,
-	})
-	if err != nil {
-		return microerror.Mask(err)
+	// This is importand key roation is enabled before creation alias.
```

Comment:
I find this wording slightly confusing. Maybe something like this:
`// IMPORTANT: Key rotation must be enabled before creating alias.`

Reply:
I added `// NOTE: Key rotation must be enabled before creating alias.`. I think `NOTE` is recognized by more editors (mine highlights it). Is it good enough?